packages/core/useFetch/index.ts

```diff
@@ -267,3 +284,43 @@ export function useFetch<T>(url: MaybeRef<string>, ...args: any[]): UseFetchRetu
 
   return shell
 }
+
+function joinPaths(start: string, end: string): string {
+  if (!start.endsWith('/') && !end.startsWith('/'))
+    return `${start}/${end}`
+
+  return `${start}${end}`
+}
+
+export function createFetch(config: CreateFetchOptions) {
+  let options = config.options || {}
+  let fetchOptions = config.fetchOptions || {}
+
+  function useFactoryFetch(url: MaybeRef<string>, ...args: any[]) {
+    const computedUrl = computed(() => joinPaths(unref(config.baseUrl), unref(url)))
+
+    // Merge properties into a single object
+    if (args.length > 0) {
+      if ('immediate' in args[0] || 'refetch' in args[0]) { options = { ...options, ...args[0] } }
```

Comment:
Not sure about this check, as we are likely to have more options in the future. Or maybe we can have a utils function that can be reuse as we are doing it four times now.

Reply:
Hmm yeah I was thinking about this as well. I'm not sure exactly what a clean solution might be for this. Off the top of my head perhaps like you said we could have a util function to handle this. Maybe something like

```ts
function hasProps(obj: object, props: string[]) {
  return Object.keys(obj).filter(key => props.includes(key)).length > 0
}

hasProps(args[0], ['immediate', 'refetch'])
```

In this case we would still need to update the parameters when we add more options, but it is a little cleaner than before. What do you think?

Reply:
I am thinking about this

```ts
// in shared utils
function containsProp(obj: object, ...props: string[]) {
  return props.some(k => k in obj)
}

// in this file
function isFetchOptions(obj: object): obj is UseFetchOptions {
  return containsProp(obj, 'immediate', 'refetch')
}

isFetchOptions(args[0])
```

We can also add a comment near `UseFetchOptions` saying remember to update that function

Reply:
That looks good to me, I'll add it 😄

Reply:
For now I put the `containsProp` function in `shared/utils/index.ts`. If you'd like it moved to a different file or want it in a new file just let me know